### ISSUE TITLE
Near-total system hang, and lock screen failed to appear, upon resume

### ISSUE THREAD

**Issue by LinuxOnTheDesktop** [linuxmint/cinnamon-screensaver#359]
```
 * Cinnamon-Screensaver version (cinnamon-screensaver --version) 4.8.1
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...) Mint 20.1
 * Graphics hardware *and* driver used: Intel; Mesa
 * 32 or 64 bit: 64
 ```
[xsesh.txt](https://github.com/linuxmint/cinnamon-screensaver/files/5826319/xsesh.txt)
[dmesg.txt](https://github.com/linuxmint/cinnamon-screensaver/files/5826322/dmesg.txt)

**Issue**

When the laptop resumed I could do nothing except open a new tty and the lock screen which should have been showing was not showing.

**Steps to reproduce**

Unclear. Perhaps merely (1) suspend and then (2) resume PC.

**Expected behaviour**

Lock-screen. No hang.

**Other information**

Caused by (the fix for) #354? For, I have not had the problem before, at least not for a long time.

I note also that the xsession-errors contains a lot about the network applet.

**Neofetch**

OS: Linux Mint 20.1 x86_64 
Host: 20KH001UKP ThinkPad X1 Carbon 6th 
Kernel: 5.8.18-050818-generic 
Uptime: 7 mins 
Packages: 3325 (dpkg) 
Shell: bash 5.0.17 
Resolution: 2560x1440 @ 60.01Hz 
DE: Cinnamon 
WM: Mutter (Muffin) 
WM Theme: Mint-Y-Dark (Mint-Y-Dark) 
Theme: Mint-Y-Dark [GTK2/3] 
Icons: Paper [GTK2/3] 
Terminal: xfce4-terminal 
Terminal Font: InconsolataGo 12 
CPU: Intel i7-8550U (8) @ 4.000GHz 
GPU: Intel UHD Graphics 620 
Memory: 2391MiB / 15899MiB 

**[Uploaded info from the Mint System Info tool](https://termbin.com/oerii)**.

**Comment by mtwebster** [comment]
> Caused by (the fix for) #354?
I can't imagine this being the case - completely unrelated.

Do you have any system-monitor-type applets or desklets active? I see conky, is that new or been there?

I'm more inclined to consider [this](https://github.com/linuxmint/cjs/commit/1210c910935e4e3ab5335725a34adb8f661a8d95) but without more data or reproduction I can't say more. Are you on cjs 4.8.2?

**Comment by LinuxOnTheDesktop** [comment]
> Do you have any system-monitor-type applets or desklets active? I see conky, is that new or been there?

I have the (third-party) Battery Monitor Applet. I do have a Conky. The Conky runs about a dozen programs - some C, some Lua - and executes (directly) a few Bash commands; but I do not think I have changed any of that recently.

Also - it may prove worth my detailing this - I have as Dash script that runs on resume. That script works around various problems. To wit: (1) some power-saving settings are not restored on resume; (2) wifi is not re-enabled if Ethernet has been removed during sleep; (3) sometimes my wifi network does not get reconnected to - even when wifi is on - after sleep. I made a small change to that latter script recently, come to think of it. The change in question was in service of 3. For, I found that, as of using Mint 20.1, sometimes I had problem 3. So I made the script do a `sleep` before it did `nmcli r wifi on` and I put that command in a subshell. But I think I got the subshell syntax wrong: I used curly brackets (`{`) rather than parentheses (`(`) but surely that would not or at least should not cause a hang.

> I can't imagine this being the case - completely unrelated.

I thought there might be a relation because I took it that the fix for the security problem involved making Cinnamon crash more readily.

**Comment by leigh123linux** [comment]
The issue doesn't reproduce here with new cjs.

```
$ rpm -q cjs
cjs-4.8.2-1.fc33.x86_64
```
```
$ inxi -GSC
System:
  Host: leigh Kernel: 5.10.6-200.fc33.x86_64 x86_64 bits: 64 
  Desktop: Cinnamon 4.8.6 Distro: Fedora release 33 (Thirty Three) 
CPU:
  Info: 6-Core model: AMD Ryzen 5 3600 bits: 64 type: MT MCP 
  L2 cache: 3072 KiB 
  Speed: 2198 MHz min/max: 2200/3600 MHz Core speeds (MHz): 1: 2200 2: 2202 
  3: 2057 4: 2052 5: 2198 6: 3602 7: 2058 8: 2055 9: 2197 10: 2193 11: 2195 
  12: 3598 
Graphics:
  Device-1: NVIDIA GP107 [GeForce GTX 1050 Ti] driver: nvidia v: 460.32.03 
  Display: x11 server: Fedora Project X.org 1.20.10 driver: nvidia 
  unloaded: fbdev,modesetting,nouveau,vesa resolution: 3840x2160~60Hz 
  OpenGL: renderer: GeForce GTX 1050 Ti/PCIe/SSE2 v: 4.6.0 NVIDIA 460.32.03
```

.xsession_errors is clear of errors after suspend.

```
Clutter-Message: 16:30:53.616: Sync method: PRESENTATION TIME
Cjs-Message: 16:30:53.699: JS LOG: About to start Cinnamon
Cjs-Message: 16:30:53.739: JS LOG: [LookingGlass/info] Cinnamon.AppSystem.get_default() started in 39 ms
Cjs-Message: 16:30:53.743: JS LOG: [LookingGlass/info] loading user theme: /usr/share/themes/Mint-Y-Dark-Aqua/cinnamon/cinnamon.css
Cjs-Message: 16:30:53.757: JS LOG: [LookingGlass/info] added icon directory: /usr/share/themes/Mint-Y-Dark-Aqua/cinnamon
Cjs-Message: 16:30:53.932: JS LOG: [LookingGlass/info] loaded at Sun Jan 17 2021 16:30:53 GMT+0000 (Greenwich Mean Time)
Cjs-Message: 16:30:53.932: JS LOG: Cinnamon started at Sun Jan 17 2021 16:30:53 GMT+0000 (Greenwich Mean Time)
Cjs-Message: 16:30:53.938: JS LOG: [LookingGlass/info] ExtensionSystem started in 0 ms
Cjs-Message: 16:30:53.938: JS LOG: [LookingGlass/info] DeskletManager started in 0 ms
Cjs-Message: 16:30:53.938: JS LOG: [LookingGlass/info] SearchProviderManager started in 0 ms
openGL version 3.1 detected (GL3 Cogl Driver)
Cjs-Message: 16:30:53.972: JS LOG: [LookingGlass/info] Role locked: tray
Cjs-Message: 16:30:53.973: JS LOG: [LookingGlass/info] Loaded applet systray@cinnamon.org in 15 ms
Cjs-Message: 16:30:53.989: JS LOG: [LookingGlass/info] Loaded applet show-desktop@cinnamon.org in 16 ms
St-Message: 16:30:53.993: cogl npot texture sizes SUPPORTED
Cjs-Message: 16:30:54.011: JS LOG: [LookingGlass/info] Loaded applet calendar@cinnamon.org in 22 ms
Cjs-Message: 16:30:54.019: JS LOG: [LookingGlass/info] Loaded applet keyboard@cinnamon.org in 7 ms
Cjs-Message: 16:30:54.037: JS LOG: [LookingGlass/info] Loaded applet printers@cinnamon.org in 17 ms
Cjs-Message: 16:30:54.042: JS LOG: [LookingGlass/info] Role locked: notifications
Cjs-Message: 16:30:54.045: JS LOG: [LookingGlass/info] Loaded applet notifications@cinnamon.org in 8 ms
Cjs-Message: 16:30:54.054: JS LOG: [LookingGlass/info] Loaded applet removable-drives@cinnamon.org in 8 ms
Cjs-Message: 16:30:54.062: JS LOG: [LookingGlass/info] Loaded applet power@cinnamon.org in 8 ms
Cjs-Message: 16:30:54.065: JS LOG: [LookingGlass/info] Loaded applet xapp-status@cinnamon.org in 2 ms
Cjs-Message: 16:30:54.080: JS LOG: [LookingGlass/info] Role locked: panellauncher
Cjs-Message: 16:30:54.082: JS LOG: [LookingGlass/info] Loaded applet grouped-window-list@cinnamon.org in 17 ms

** (csd-power:1787): CRITICAL **: 16:30:54.087: abs_to_percentage: assertion 'max > min' failed
Cjs-Message: 16:30:54.105: JS LOG: [LookingGlass/info] Loaded applet sound@cinnamon.org in 23 ms
Cjs-Message: 16:30:54.121: JS LOG: [LookingGlass/info] Loaded applet network@cinnamon.org in 16 ms
Cjs-Message: 16:30:54.187: JS LOG: [LookingGlass/info] Loaded applet weather@mockturtl in 66 ms
Cjs-Message: 16:30:54.202: JS LOG: Unknown network device type, is 14
Cjs-Message: 16:30:54.209: JS LOG: Unknown network device type, is 29
Cjs-Message: 16:30:54.209: JS LOG: Unknown network device type, is 13
Cjs-Message: 16:30:54.209: JS LOG: Unknown network device type, is 16
Cjs-Message: 16:30:54.209: JS LOG: Unknown network device type, is 18
Cjs-Message: 16:30:54.277: JS LOG: [LookingGlass/info] Loaded applet menu@cinnamon.org in 90 ms
Cjs-Message: 16:30:54.277: JS LOG: [LookingGlass/info] AppletManager started in 345 ms
Cjs-Message: 16:30:54.279: JS LOG: [LookingGlass/info] Cinnamon took 580 ms to start

(cinnamon:1947): Cjs-WARNING **: 16:30:54.285: Some code called array.toString() on a Uint8Array instance. Previously this would have interpreted the bytes of the array as a string, but that is nonstandard. In the future this will return the bytes as comma-separated digits. For the time being, the old behavior has been preserved, but please fix your code anyway to explicitly call ByteArray.toString(array). (Note that array.toString() may have been called implicitly.)

ByteArray.toString(array) is *not* backward compatible with previous (< 4.8) versions of Cinnamon. You will have add versioning support to your applet if it does not already have it, so that compatibility can be maintained for users on older Cinnamon versions.

See:
https://projects.linuxmint.com/reference/git/cinnamon-tutorials/xlet-versioning.html

0 LoadContents/</<() ["/usr/share/cinnamon/js/misc/fileUtils.js line 211 > Function":1884:33]


(cinnamon:1947): Cjs-WARNING **: 16:30:54.285: Some code called array.toString() on a Uint8Array instance. Previously this would have interpreted the bytes of the array as a string, but that is nonstandard. In the future this will return the bytes as comma-separated digits. For the time being, the old behavior has been preserved, but please fix your code anyway to explicitly call ByteArray.toString(array). (Note that array.toString() may have been called implicitly.)

ByteArray.toString(array) is *not* backward compatible with previous (< 4.8) versions of Cinnamon. You will have add versioning support to your applet if it does not already have it, so that compatibility can be maintained for users on older Cinnamon versions.

See:
https://projects.linuxmint.com/reference/git/cinnamon-tutorials/xlet-versioning.html

0 LoadContents/</<() ["/usr/share/cinnamon/js/misc/fileUtils.js line 211 > Function":1885:32]

Cjs-Message: 16:30:54.285: JS LOG: [LookingGlass/info] [weather@mockturtl#13]: Saved locations are loaded in from location store at: '/home/leigh/.config/weather-mockturtl/locations.json'
Failed to rename ‘/home/leigh/.abrt/spool’ to ‘/home/leigh/.cache/abrt/spool’: No such file or directory
Failed to play sound: Sound disabled
** Message: 16:30:55.045: nemo-desktop: session is cinnamon, establishing proxy
Cjs-Message: 16:30:55.073: JS LOG: [LookingGlass/info] Hiding systray: network
Cjs-Message: 16:30:55.357: JS LOG: [LookingGlass/info] [weather@mockturtl#13]: Weather Information refreshed
CsNotificationWatcher: Could not add match rule to bus - GDBus.Error:org.freedesktop.DBus.Error.MatchRuleInvalid: Invalid match rule
cinnamon-session[1569]: WARNING: t+7.05911s: Detected that screensaver has appeared on the bus

** (csd-media-keys:1783): WARNING **: 16:31:37.787: Unable to get default sink
** Message: 16:31:56.845: Existing monitor config (/home/leigh/.config/cinnamon-monitors.xml) not found during hotplug or laptop lid event. Looking for legacy configuration (monitors.xml)
** Message: 16:33:22.531: Existing monitor config (/home/leigh/.config/cinnamon-monitors.xml) not found during hotplug or laptop lid event. Looking for legacy configuration (monitors.xml)
** Message: 16:37:18.072: Existing monitor config (/home/leigh/.config/cinnamon-monitors.xml) not found during hotplug or laptop lid event. Looking for legacy configuration (monitors.xml)
```

**Comment by LinuxOnTheDesktop** [comment]
I should have said: I have cjs '4.8.2+ulyssa' (judging by Synaptic; I do not have rpm installed).

**Comment by LinuxOnTheDesktop** [comment]
The plot thickens: well I tell my laptop to suspend (well, one of them - I have not tested the other yet), this happens:

the lockscreen appears (and, incidentally, does not tell me that the system is going into suspend - but I don't think it ever did tell me that);
the lockscreen disappears after a few seconds;
the lockscreen reappears - the PC does not go to sleep.

Two other people report the problem [here on the Mint forum](https://forums.linuxmint.com/viewtopic.php?t=339612). Should I open a new bug report.

**Comment by LinuxOnTheDesktop** [comment]
I have just experienced this problem again. How can I stop this computer from failing to awake?

```
$ inxi -GSC
System:    Host: X230 Kernel: 5.8.0-44-generic x86_64 bits: 64 Desktop: Cinnamon 4.8.6 
           Distro: Linux Mint 20.1 Ulyssa 
CPU:       Topology: Dual Core model: Intel Core i5-3320M bits: 64 type: MT MCP L2 cache: 3072 KiB 
           Speed: 2291 MHz min/max: 1200/3300 MHz Core speeds (MHz): 1: 2223 2: 2241 3: 2152 4: 2248 
Graphics:  Device-1: Intel 3rd Gen Core processor Graphics driver: i915 v: kernel 
           Display: x11 server: X.Org 1.20.9 driver: modesetting resolution: 1366x768~60Hz 
           OpenGL: renderer: Mesa DRI Intel HD Graphics 4000 (IVB GT2) v: 4.2 Mesa 20.2.6 
```

**Comment by LinuxOnTheDesktop** [comment]
The problem persists.

**Comment by LinuxOnTheDesktop** [comment]
A week or so ago one of my ThinkPads - an X230 - running Mint Cinnamon 20.3 - encountered this problem again.

**Comment by LinuxOnTheDesktop** [comment]
The X230 iteration of ThinkPads has a particular problem. (See #368.) But I continue to have the problem on a ThinkPad X1 Carbon (6th generation). It is really not very good that every so often and for no good reason an operating system locks the user out. How about some diagnosis? Is this an upstream problem, or not?

**Comment by LinuxOnTheDesktop** [comment]
How is this problem not a serious one that should be near the top of the to-fix list? I am minded to cancel my regular payment to Mint.

**Comment by leigh123linux** [comment]
> How is this problem not a serious one that should be near the top of the to-fix list? I am minded to cancel my regular payment to Mint.

You can expect kernel issues to last years on a LTS based distro, have you filed a issue against the ubuntu kernel?

**Comment by LinuxOnTheDesktop** [comment]
It does seem (as I think I commented somewhere else on the Mint bug tracker) that _ThinkPads of the model X230 do not to awake properly from sleep because of a **BIOS** problem_ - or perhaps it is fair to say: because of a BIOS problem that the Linux kernel, or Ubuntu, has failed to work around.

_My other ThinkPad, an X1CG6, also had the problem_. I do not think that _that_ problem owed to BIOS or to the kernel or to Ubuntu. _Yet, since I installed M. Webster's latest improvements_ (described [here](https://github.com/linuxmint/cinnamon-screensaver/issues/416#issuecomment-1256233029)), which I did some weeks ago, _resumption from sleep has not hung the desktop on that PC_. _I infer that those improvements have solved the problem_ (insofar as it was a Cinnamon problem). So, I'm closing this report.
